torch/nn/functional.py

```diff
@@ -4568,7 +4568,9 @@ def interpolate(  # noqa: F811
         # Two levels are necessary to prevent TorchScript from touching
         # are_deterministic_algorithms_enabled.
         if not torch.jit.is_scripting():
-            if torch.are_deterministic_algorithms_enabled() and input.is_cuda:
+            if torch.are_deterministic_algorithms_enabled() and (
+                input.is_cuda or input.is_xpu
```

Comment:
>> Is this check needed at the first place? I mean I wonder whether input.is_cuda() condition should be just removed instead?
>
> For other backend, like cpu, whose kernel is deterministic implementation. So there is not need to call decompose implementation, which has slow performance.

I still think it's better to have some other check here instead of explicit iterations over specific backends. Dedicated backend attribute to check might be a better variant. However, that's something to decide on by pytorch maintainers. So, I'll approve the PR from my side and leave that to maintainers.